Cql/Cql.CqlToElm/TypeConverter.cs

```diff
@@ -0,0 +1,426 @@
+﻿using Hl7.Cql.CqlToElm.Builtin;
+using Hl7.Cql.Elm;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Hl7.Cql.CqlToElm
+{
+    /// <summary>
+    /// Implements type conversion according to the rules defined by the specification. 
+    /// </summary>
+    /// <seealso href="https://cql.hl7.org/03-developersguide.html#conversion-precedence"/>
+    internal class TypeConverter
```

Comment:
I asked ChatGPT:
> User
what is the process of changing one argument's type to another when binding an argument to an operand?

> The process of changing one argument's type to another when binding an argument to an operand in programming languages, particularly in the context of function calls or operations, is known as **type coercion** or **type conversion**. This process can be either implicit or explicit.

Since `TypeConverter` is taken on the C# side, I'll use `Coerce` as the verb instead of `Convert`.  Generally, when I name classes, particularly injected services I think about the main verb.  In this case the key function is called `Convert` (to be renamed `Coerce`).